Doppler.ImageAnalyzer.Api/Services/MongoDB/ImageAnalyzerMongoDBContextSettings.cs

```diff
@@ -1,9 +1,37 @@
-namespace Doppler.PushContact.Services
+using MongoDB.Driver;
+
+namespace Doppler.ImageAnalyzer.Api.Services.MongoDB
 {
     public class ImageAnalyzerMongoDBContextSettings
     {
-        public string ConnectionString { get; set; }
-        public string DatabaseName { get; set; }
-        public string Password { get; set; }
+        public string? ConnectionString { get; set; }
+
+        private string? _databaseName;
+        public string? DatabaseName
+        {
+            get
+            {
+                var mongoUrl = new MongoUrl(ConnectionString);
+                return mongoUrl.DatabaseName ?? _databaseName;
+            }
+            set
+            {
+                _databaseName = value;
+            }
+        }
+
+        private string? _password;
+        public string? Password
+        {
+            get
+            {
+                var mongoUrl = new MongoUrl(ConnectionString);
+                return mongoUrl.Password ?? _password;
+            }
+            set
+            {
+                _password = value;
+            }
+        }
```

Comment:
I suggest keeping it as a POCO class and renaming it as `ImageAnalyzerDBContextSettings` allowing use with other database implementations in the future

Reply:
I think that this is related to what we are discussing [here](https://github.com/FromDoppler/doppler-image-analyzer/pull/300#discussion_r1344199000), do you prefer to talk about this?